### PR TITLE
Replace beautysh hook with cisagov/pre-commit-shfmt

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR replaces the [beautysh](https://github.com/lovesegfault/beautysh) [pre-commit](https://pre-commit.com) hook with one that leverages [shfmt](https://github.com/mvdan/sh#shfmt) instead.

This PR relies on https://github.com/cisagov/setup-env-github-action/pull/33 being approved and merged before the Actions workflow will pass.

## 💭 Motivation and context ##

We have run into some issues where `beautysh` chokes on bash scripts and will not correctly process them with its own formatting generated resulting in errors like `error: indent/outdent mismatch: 2.`. After doing research for an alternative I settled on `shfmt` as a mature project that could be leveraged and wrote [cisagov/pre-commit-shfmt](https://github.com/cisagov/pre-commit-shfmt) to use it as a hook.

## 🧪 Testing ##

I tested it against code that we have had to use the `# @formatter: off` directive for `beautysh` in to successfully pass linting checks.

In [cisagov/cool-assessment-terraform](https://github.com/cisagov/cool-assessment-terraform/blob/2355d47745b361cae1a135d8a60dfe0be0630d6e/cloud-init/map-hostname-to-localhost.tpl.sh#L40-L47):

```diff
diff --git a/cloud-init/map-hostname-to-localhost.tpl.sh b/cloud-init/map-hostname-to-localhost.tpl.sh
index e7de0a2..d912804 100644
--- a/cloud-init/map-hostname-to-localhost.tpl.sh
+++ b/cloud-init/map-hostname-to-localhost.tpl.sh
@@ -48,8 +48,7 @@ grep --quiet --ignore-case "^127\.0\.0\.1\s$${escaped_hostname}\s*$" "${hosts_fi
 grep_rc="$?"
 set -o errexit
 
-if [[ "$grep_rc" -ne 0 ]]
-then
+if [[ "$grep_rc" -ne 0 ]]; then
   # Add the mapping to the hosts file
   echo -e "\n127.0.0.1 ${hostname}" >> "${hosts_file}"
 else
 ```

In [cisagov/gatherer](https://github.com/cisagov/gatherer/blob/82876bcb531c3111daca5cdf72c0405f7b7a8268/src/gather-domains.sh#L95-L101):

```diff
diff --git a/src/gather-domains.sh b/src/gather-domains.sh
index 88b37c0..37126ea 100755
--- a/src/gather-domains.sh
+++ b/src/gather-domains.sh
@@ -7,8 +7,7 @@ HOME_DIR=/home/cisa
 OUTPUT_DIR=$HOME_DIR/shared/artifacts
 
 # Create the output directory, if necessary
-if [ ! -d $OUTPUT_DIR ]
-then
+if [ ! -d $OUTPUT_DIR ]; then
   mkdir $OUTPUT_DIR
 fi
 
@@ -43,7 +42,7 @@ wget https://raw.githubusercontent.com/cisagov/scan-target-data/develop/current-
 tail -n +2 $OUTPUT_DIR/current-federal-non-dotgov.csv > \
   /tmp/current-federal-non-dotgov.csv
 cat $OUTPUT_DIR/current-federal.csv \
-  /tmp/current-federal-non-dotgov.csv  > \
+  /tmp/current-federal-non-dotgov.csv > \
   $OUTPUT_DIR/current-federal_modified.csv
 ###
 # Remove the FED.US domain.  This is really a top-level domain,
@@ -93,11 +92,11 @@ sed -i '/[^,]*,[^,]*,Library of Congress,/d;/[^,]*,[^,]*,Government Publishing O
 # lovesegfault/beautysh#82 to document this.
 #
 # @formatter:off
-TOP_LEVEL_DOMAINS=$(cut --delimiter=, --fields=1 $OUTPUT_DIR/current-federal_modified.csv | \
-  awk -F"." '{if (NR > 1) print "."$NF;}' | \
-  sort --ignore-case --unique | \
-  tr "[:upper:]" "[:lower:]" | \
-  paste --serial --delimiters=,)
+TOP_LEVEL_DOMAINS=$(cut --delimiter=, --fields=1 $OUTPUT_DIR/current-federal_modified.csv \
+  | awk -F"." '{if (NR > 1) print "."$NF;}' \
+  | sort --ignore-case --unique \
+  | tr "[:upper:]" "[:lower:]" \
+  | paste --serial --delimiters=,)
 # @formatter:on
 $HOME_DIR/domain-scan/gather current_federal,analytics_usa_gov,censys_snapshot,rapid,eot_2012,eot_2016,cyhy,other \
   --suffix="$TOP_LEVEL_DOMAINS" --ignore-www --include-parents \
@@ -114,7 +113,7 @@ cp results/gathered.csv gathered.csv
 cp results/gathered.csv $OUTPUT_DIR/gathered.csv
 
 # Remove extra columns
-cut -d"," -f1 gathered.csv  > scanme.csv
+cut -d"," -f1 gathered.csv > scanme.csv
 
 # Remove characters that might break parsing
 sed -i '/^ *$/d;/@/d;s/ //g;s/\"//g;s/'\''//g' scanme.csv
```

In [cisagov/sensitive-data-scanner](https://github.com/cisagov/sensitive-data-scanner/blob/first-commits/src/run-gitleaks.sh):

```diff
diff --git a/src/run-gitleaks.sh b/src/run-gitleaks.sh
index 365f0b7..53e5cb1 100755
--- a/src/run-gitleaks.sh
+++ b/src/run-gitleaks.sh
@@ -7,7 +7,8 @@ set -o nounset
 # set -o errexit
 set -o pipefail
 
-USAGE=$(cat << EOF
+USAGE=$(
+  cat << EOF
 Run gitleaks over multiple GitHub repositories.
 
 Usage:
@@ -38,18 +39,18 @@ SKIP_TO_LINE=0
 THREADS=2
 
 # Parse command line arguments
-while (( "$#" )); do
+while (("$#")); do
   case "$1" in
-    -h|--help)
+    -h | --help)
       echo "${USAGE}" >&2
       exit 0
       ;;
-    -s|--skipto)
+    -s | --skipto)
       shift
       SKIP_TO_LINE=$1
       shift
       ;;
-    -t|--threads)
+    -t | --threads)
       shift
       THREADS=$1
       shift
@@ -69,8 +70,8 @@ done
 eval set -- "$PARAMS"
 
 if [ "$#" -ne 1 ]; then
-    echo "${USAGE}"
-    exit 1
+  echo "${USAGE}"
+  exit 1
 fi
 
 URL_FILE=$1
@@ -82,7 +83,7 @@ log "Starting scan at ${timestamp} using ${THREADS} threads."
 line_number=0
 while read -r url; do
   line_number=$((line_number + 1))
-  if (( line_number < SKIP_TO_LINE )); then
+  if ((line_number < SKIP_TO_LINE)); then
     continue
   fi
   log "Processing URL #${line_number}: ${url}"
```

I am happy to do any additional testing requested to make sure we're all satisfied with this changeover.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Excellent work- thanks for taking the initiative to do this! ⚡

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -48,13 +48,20 @@ repos:
...
-  - repo: https://github.com/lovesegfault/beautysh
-    rev: v6.1.0
+  - repo: https://github.com/cisagov/pre-commit-shfmt
+    rev: v0.0.2
     hooks:
-      - id: beautysh
+      - id: shfmt
         args:
-          - --indent-size
+          # Indent by two space
```
Comment (jsf9k) [inline .pre-commit-config.yaml]:
```suggestion
          # Indent by two spaces
```

**Review by jsf9k — APPROVED** [review]
I made one extremely minor suggestion for a comment.

Reminder that we will have to search for `@formatter` in the resulting Lineage PRs after we merge this.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mcdonnnj) [reply .pre-commit-config.yaml]: I rebased to fix this and it now lives in 406b688.

**Review by jsf9k — APPROVED** [review]

**Pull request merged**